lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -84,64 +404,57 @@ function collectAllStringsInDir(dir, strings = {}) {
           let lastPropertyEndIndex = 0;
           for (const property of stmt.declarations[0].init.properties) {
             const key = property.key.name;
-            const message = exportVars.UIStrings[key];
-            const description = computeDescription(ast, property, lastPropertyEndIndex);
-            strings[`${relativePath} | ${key}`] = {message, description};
-            lastPropertyEndIndex = property.range[1];
-          }
-        }
-      }
-    }
-  }
+            const val = exportVars.UIStrings[key];
+            const {description, examples} = computeDescription(ast, property, val, lastPropertyEndIndex);
 
-  return strings;
-}
+            const converted = convertMessageToPlaceholders(val, examples);
 
-/**
- * @param {Record<string, ICUMessageDefn>} strings
- */
-function createPsuedoLocaleStrings(strings) {
-  /** @type {Record<string, ICUMessageDefn>} */
-  const psuedoLocalizedStrings = {};
-  for (const [key, defn] of Object.entries(strings)) {
-    const message = defn.message;
-    const psuedoLocalizedString = [];
-    let braceCount = 0;
-    let useHatForAccentMark = true;
-    for (const char of message) {
-      psuedoLocalizedString.push(char);
-      if (char === '{') {
-        braceCount++;
-      } else if (char === '}') {
-        braceCount--;
-      }
+            const messageKey = `${relativePath} | ${key}`;
 
-      // Hack to not change {plural{ICU}braces} nested an odd number of times.
-      // ex: "{itemCount, plural, =1 {1 link found} other {# links found}}"
-      // becomes "{itemCount, plural, =1 {1 l̂ín̂ḱ f̂óûńd̂} other {# ĺîńk̂ś f̂óûńd̂}}"
-      // ex: "{itemCount, plural, =1 {1 link {nested_replacement} found} other {# links {nested_replacement} found}}"
-      // becomes: "{itemCount, plural, =1 {1 l̂ín̂ḱ {nested_replacement} f̂óûńd̂} other {# ĺîńk̂ś {nested_replacement} f̂óûńd̂}}"
-      if (braceCount % 2 === 1) continue;
+            /** @type {ICUMessageDefn} */
+            const icuDefn = {
+              message: converted.message,
+              description,
+            };
 
-      // Add diacritical marks to the preceding letter, alternating between a hat ( ̂) and an acute (´).
-      if (/[a-z]/i.test(char)) {
-        psuedoLocalizedString.push(useHatForAccentMark ? `\u0302` : `\u0301`);
-        useHatForAccentMark = !useHatForAccentMark;
+            if (converted.placeholders.constructor === Object) {
```

Comment:
check still not needed? It's either `Record<string, ICUPlaceholderDefn>` or `undefined`, either way fine to assign to `icuDefn.placeholders`

Reply:
We tracked this down, turns out we were both wrong.